File: Data Structures & Algorithms/two-integer-sum/submission-67.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        has = {}
        for i,j in enumerate(nums):
            needed_sum = target - j
            if needed_sum in has:
                return [has[needed_sum],i]

            has[j] = i  ## J is the number itself and we need to add it to the index, if we the number is not in the dictionary or hash map
        return []

File: Data Structures & Algorithms/duplicate-integer/submission-67.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        for(int i = 0; i < size(nums); i++){
            for (int j = i + 1; j < size(nums); j++){
                if (nums[i] == nums[j]){
                    return true;
                }
                

            }

         
        }
           return false; 

        
    }
};